src/string.cr

```diff
@@ -1,6 +1,8 @@
 require "c/stdlib"
 require "c/string"
 
+require "regex/*"
```

Comment:
Is this added because of the order in the prelude file?

Reply:
yes, without that referring to `Regex::MatchData` would fail :/

Reply:
I think This is should be added at the prelude level.
Since string is required in prelude we might as well make it explicit at that level.

Reply:
I don't think this can be accepted. `string` is a core type. It must not depend on `regex`.

Reply:
Regex is also a core type. It has specialized syntax for it. There's `$1`, `$~`, etc.

Reply:
It looks like `String` at master already refers to `Regex::MatchData` [here](https://github.com/crystal-lang/crystal/pull/8162/files#diff-87ebb896824e7dc3bd8d5b29293bcf00L3737).  Is there a way why this is different and needs to require `regex/*`

Reply:
I don't know why, but without this require the code doesn't compile.

Reply:
What is the error?

Reply:
```
./bin/crystal spec spec/std/string_spec.cr --error-trace
Using compiled compiler at `.build/crystal'
error in line 1
Error: while requiring "prelude"


In src/prelude.cr:29:1

 29 | require "string"
      ^
Error: while requiring "string"


In src/string.cr:3735:22

 3735 | include Iterator(Regex::MatchData)
                         ^---------------
Error: undefined constant Regex::MatchData
```